dbfit-java/informix/src/test/java/dbfit/environment/InformixBigDecimalTransformerTest.java

```diff
@@ -0,0 +1,42 @@
+package dbfit.environment;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.lang.Double;
+import java.sql.SQLException;
+
+import dbfit.util.NormalisedBigDecimal;
+
+public class InformixBigDecimalTransformerTest {
+
+    private final InformixBigDecimalTransformer bdt = new InformixBigDecimalTransformer();
+    private final BigDecimal inBigDec = new BigDecimal("123.45");
+
+    @Test
+    public void transformsBigDecimaltoBigDecimalTest() throws SQLException {
+        BigDecimal bd = new BigDecimal(inBigDec.toString());
+        BigDecimal outBigDec = (BigDecimal) bdt.transform(bd);
+        assertEquals(outBigDec, inBigDec);
+    }
+
+    @Test
+    public void transformsNormalisedBigDecimaltoBigDecimalTest() throws SQLException {
+        NormalisedBigDecimal nbd = new NormalisedBigDecimal(inBigDec);
+        BigDecimal outBigDec = (BigDecimal) bdt.transform(nbd);
+        assertEquals(outBigDec, inBigDec);
+    }
+
+    @Test
+    public void rejectsNonNormalisedBigDecimalInputTest() {
```

Comment:
I did take a look at `@Rule` and `expect` and `expectMessage`. I was thinking of something like this: -

```java
    @Rule
    public ExpectedException exception = ExpectedException.none();

    @Test
    public void rejectsNonNormalisedBigDecimalInputTest() {
        Double notABigBecimal = 1.0;
        exception.expect(java.sql.SQLException.class);
        exception.expectMessage("InformixBigDecimalTransformer cannot transform objects of type java.lang.Double");
        bdt.transform(notABigBecimal);
    }
```

I'm not quite sure how this `@Rule` annotation works. Does it set the expected exception (to *no exception*) for all of the test cases in the test class, or just those that follow it in the order of the code?

Also, because `TypeTransformer.transform()` specifies that `SQLException` is thrown then I need a `try...catch` block around the `transform()` call too and this doesn't look as good as it might: -

```java
    @Rule
    public ExpectedException exception = ExpectedException.none();

    @Test
    public void rejectsNonNormalisedBigDecimalInputTest() {
        Double notABigBecimal = 1.0;
        exception.expect(java.sql.SQLException.class);
        exception.expectMessage("InformixBigDecimalTransformer cannot transform objects of type java.lang.Double");
        try {
            bdt.transform(notABigBecimal);
        } catch (java.sql.SQLException e) {
        }
    }
```